tracer/src/Datadog.Trace/OTelMetrics/OtlpExporter.cs

```diff
@@ -0,0 +1,468 @@
+// <copyright file="OtlpExporter.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#if NET6_0_OR_GREATER
+
+using System;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Telemetry;
+
+#nullable enable
+
+namespace Datadog.Trace.OTelMetrics
+{
+    /// <summary>
+    /// OTLP Exporter implementation that exports metrics using the OpenTelemetry Protocol.
+    /// This is the concrete implementation of the MetricExporter base class that handles OTLP protocol specifics.
+    /// Supports both grpc and http/protobuf transports as specified in the RFC.
+    /// This is a Push Metric Exporter that sends metrics via the OpenTelemetry Protocol.
+    /// </summary>
+    internal class OtlpExporter : MetricExporter
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(OtlpExporter));
+        private readonly HttpClient _httpClient;
+        private readonly Telemetry.Metrics.MetricTags.Protocol _protocolTag;
+        private readonly OtlpMetricsSerializer _serializer;
+        private readonly Uri _endpoint;
+        private readonly IReadOnlyDictionary<string, string> _headers;
+        private readonly int _timeoutMs;
+        private readonly Configuration.OtlpProtocol _protocol;
+
+        public OtlpExporter(Configuration.TracerSettings settings)
+        {
+            _endpoint = settings.OtlpMetricsEndpoint;
+            _headers = settings.OtlpMetricsHeaders;
+            _timeoutMs = settings.OtlpMetricsTimeoutMs;
+            _protocol = settings.OtlpMetricsProtocol;
+
+            _protocolTag = _protocol switch
+            {
+                Configuration.OtlpProtocol.Grpc => Telemetry.Metrics.MetricTags.Protocol.Grpc,
+                Configuration.OtlpProtocol.HttpProtobuf => Telemetry.Metrics.MetricTags.Protocol.HttpProtobuf,
+                Configuration.OtlpProtocol.HttpJson => Telemetry.Metrics.MetricTags.Protocol.HttpJson,
+                _ => Telemetry.Metrics.MetricTags.Protocol.HttpProtobuf
+            };
+            _serializer = new OtlpMetricsSerializer(settings);
+
+            // Enable h2c (HTTP/2 without TLS) for gRPC only, and only for non-HTTPS endpoints
+            if (_protocol == Configuration.OtlpProtocol.Grpc && _endpoint.Scheme != Uri.UriSchemeHttps)
+            {
+                AppContext.SetSwitch("System.Net.Http.SocketsHttpHandler.Http2UnencryptedSupport", true);
```

Comment:
This seems worthy of writing an Information level log to indicate that we're enabling this runtime switch